frontend/src/scenes/funnels/Funnel.tsx

```diff
@@ -4,21 +4,30 @@ import { ChartParams, FunnelVizType } from '~/types'
 import { FunnelBarGraph } from './FunnelBarGraph'
 import { FunnelHistogram } from './FunnelHistogram'
 import { funnelLogic } from './funnelLogic'
+import { FEATURE_FLAGS } from 'lib/constants'
+import { featureFlagLogic } from 'lib/logic/featureFlagLogic'
+import { FunnelViz } from 'scenes/funnels/FunnelViz'
 
 export function Funnel(props: Omit<ChartParams, 'view'>): JSX.Element | null {
+    const { featureFlags } = useValues(featureFlagLogic)
     const logic = funnelLogic({ dashboardItemId: props.dashboardItemId, filters: props.filters })
-    const { timeConversionBins, filters } = useValues(logic)
+    const { isValidFunnel, filters } = useValues(logic)
     const { loadResults } = useActions(logic)
 
     useEffect(() => {
         loadResults()
     }, [])
 
-    const funnel_viz_type = filters.funnel_viz_type || props.filters.funnel_viz_type
+    if (featureFlags[FEATURE_FLAGS.FUNNEL_BAR_VIZ] && isValidFunnel) {
```

Comment:
Not entirely sure about the `isValidFunnel` check here, invalid funnels should be handled by the new components too, but it seems to work so it can be a later fix.

Reply:
Removed